softhddevice.cpp

```diff
@@ -323,20 +293,40 @@ void cSoftHdDevice::HandlePause(void)
  *
  * @param event     The event to process (variant type containing specific event data)
  */
-void cSoftHdDevice::OnEventReceived(const Event& event) {
+void cSoftHdDevice::OnEventReceived(const Event& event)
+{
+	m_mutex.lock();
```

Comment:
You could use a lock guard, then unlock() is automatically called when the scope is left (e.g. during an early return). In this case it doesn't matter, but the docs of std::mutex.unlock() have a note that this method is not recommended to be used directly, but for example with a lock guard.
```suggestion
	std::lock_guard<std::mutex> lock(m_mutex);
```

Reply:
done